neo/rawio/axonarawio.py

```diff
@@ -333,6 +339,100 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
 
+    def _get_mock_analogsignal_chunk(self, channel_ids, start_frame, end_frame,
+                                     noise_std=0, return_scaled=False):
+        # Create mock continuous recording from tetrode file (.X) data
+        # by filling Gaussian noise into gaps between waveforms.
+
+        timebase_sr = int(self.file_parameters['unit']['timebase'].split(' ')[0])
+        samples_pre = int(self.file_parameters['set']['file_header']['pretrigSamps'])
+        samples_post = int(self.file_parameters['set']['file_header']['spikeLockout'])
+        sampling_rate = int(self.file_parameters['set']['sampling_rate'])
+
+        tcmap = self._get_tetrode_channel_table(channel_ids)
+
+        traces = noise_std * np.random.randn(len(channel_ids), end_frame - start_frame)
+        if return_scaled:
+            traces = traces.astype(np.float32)
+        else:
+            traces = traces.astype(np.int8)
+
+        # Loop through tetrodes and include requested channels in traces
+        itrc = 0
+        for tetrode_id in np.unique(tcmap[:, 0]):
+
+            channels_oi = tcmap[tcmap[:, 0] == tetrode_id, 2]
+
+            waveforms = self._get_spike_raw_waveforms(
+                block_index=0, seg_index=0,
+                unit_index=tetrode_id - 1,  # Tetrodes IDs are 1-indexed
+                t_start=start_frame / sampling_rate,
+                t_stop=end_frame / sampling_rate
+            )
+            waveforms = waveforms[:, channels_oi, :]
+            nch = len(channels_oi)
+
+            spike_train = self._get_spike_timestamps(
+                block_index=0, seg_index=0,
+                unit_index=tetrode_id - 1,
+                t_start=start_frame / sampling_rate,
+                t_stop=end_frame / sampling_rate
+            )
+
+            # Fill waveforms into traces timestamp by timestamp
+            for t, wf in zip(spike_train, waveforms):
+
+                t = int(t // (timebase_sr / sampling_rate))  # timestamps are at higher frequency
+                t = t - start_frame
+                if (t - samples_pre < 0) and (t + samples_post > traces.shape[1]):
+                    traces[itrc:itrc + nch, :] = \
+                        wf[:, samples_pre - t:traces.shape[1] - (t - samples_pre)]
+                elif t - samples_pre < 0:
+                    traces[itrc:itrc + nch, :t + samples_post] = wf[:, samples_pre - t:]
+                elif t + samples_post > traces.shape[1]:
+                    traces[itrc:itrc + nch, t - samples_pre:] = \
+                        wf[:, :traces.shape[1] - (t - samples_pre)]
+                else:
+                    traces[itrc:itrc + nch, t - samples_pre:t + samples_post] = wf
+
+            itrc += nch
+
+        return traces.transpose()
+
+    def _get_tetrode_channel_table(self, channel_ids):
+        '''Create auxiliary np.array with the following columns:
+        Tetrode ID, Channel ID, Channel ID within tetrode
+        This is useful in `get_traces()`
```

Comment:
I don't think the `get_traces` function you are referring to here is within Neo, so it would be better to remove the comment here to avoid confusion.

Reply:
Yes, good point!